### DISCUSSION TITLE
[Config Support]: ValueError: Camera X has a required zone for alerts Y that is not defined.

### DISCUSSION THREAD

**Discussion by luzik** [blakeblackshear/frigate#13265] [Config Support] [answered]
### Describe the problem you are having

In global configuration I've set required_zones for review. Not all cameras include this zone in its configuration. Frigate can't start with this error as below.

IMHO, when I set this option globally if specific camera do not include required_zone then it shouldn't be included in alerts|detections, but configuration is OK.

BTW..
1. Configuration check should be run before restart from GUI, just like in HA - this is awesome feature, to know that my changes in  configuration are bad before restart.
2. There is also a glitch in configuration flow. When I've edited configuration file using frigate GUI, after that  I was able to add new zone to camera. All my changes in camera zones was lost because I didn't restart after configuration change. Solution - disable all button's that affecting configuration. Solution 2 - make adding zones in GUI (after changing configuration file without reboot) merge-able.
3.  When there is syntax error in configuration editor - pressing save button gives you an error window - this is great, but configuration file is now cuted to place where error is, rest of file is lost, so I have to reload configuration editor page

### Version

0.14.0

### Frigate config file

```yaml
review:
 alerts:
   required_zones:
     - Y
cameras:
  X:
    zones:
      Z:
        coordinates: 772,334,430,313,142,284,144,344,238,354,317,377,449,402,606,412,774,423
  W:
    zones:
      Y:
        coordinates: 772,334,430,313,142,284,144,344,238,354,317,377,449,402,606,412,774,423
```

### Relevant log output

```shell
frigate                   | 2024-08-22 12:47:31.407228364  *************************************************************
frigate                   | 2024-08-22 12:47:31.407232121  *************************************************************
frigate                   | 2024-08-22 12:47:31.407235898  ***    Your config file is not valid!                     ***
frigate                   | 2024-08-22 12:47:31.407237271  ***    Please check the docs at                           ***
frigate                   | 2024-08-22 12:47:31.407238082  ***    https://docs.frigate.video/configuration/index     ***
frigate                   | 2024-08-22 12:47:31.407239184  *************************************************************
frigate                   | 2024-08-22 12:47:31.407239806  *************************************************************
frigate                   | 2024-08-22 12:47:31.407257719  ***    Config Validation Errors                           ***
frigate                   | 2024-08-22 12:47:31.407258501  *************************************************************
frigate                   | 2024-08-22 12:47:31.407259122  Camera tablet has a required zone for alerts podjazd that is not defined.
frigate                   | 2024-08-22 12:47:31.407727557  Traceback (most recent call last):
frigate                   | 2024-08-22 12:47:31.407729080    File "/opt/frigate/frigate/app.py", line 645, in start
frigate                   | 2024-08-22 12:47:31.407729851      self.init_config()
frigate                   | 2024-08-22 12:47:31.407730442    File "/opt/frigate/frigate/app.py", line 140, in init_config
frigate                   | 2024-08-22 12:47:31.407730923      self.config = user_config.runtime_config(self.plus_api)
frigate                   | 2024-08-22 12:47:31.407731454    File "/opt/frigate/frigate/config.py", line 1584, in runtime_config
frigate                   | 2024-08-22 12:47:31.407731925      verify_required_zones_exist(camera_config)
frigate                   | 2024-08-22 12:47:31.407732426    File "/opt/frigate/frigate/config.py", line 1291, in verify_required_zones_exist
frigate                   | 2024-08-22 12:47:31.407732857      raise ValueError(
frigate                   | 2024-08-22 12:47:31.407733348  ValueError: Camera X has a required zone for alerts Y that is not defined.
frigate                   | 2024-08-22 12:47:31.407733779  
frigate                   | 2024-08-22 12:47:31.407734280  *************************************************************
frigate                   | 2024-08-22 12:47:31.407734760  ***    End Config Validation Errors                       ***
frigate                   | 2024-08-22 12:47:31.407735562  *************************************************************
frigate                   | 2024-08-22 12:47:32.034953842  [INFO] Service Frigate exited with code 1 (by signal 0)
```

### Operating system

Other Linux

### Install method

Docker Compose

### Object Detector

Coral

**Comment by hawkeye217** [comment]
This is covered in the [FAQ](https://github.com/blakeblackshear/frigate/discussions/12724). Upgrade issues question 7.

**Comment by luzik** [comment]
It is not after upgrade. I've followed https://docs.frigate.video/configuration/reference - global "review" section. I've just added this section to my configuration.
My goal was to get alerts only from zones mentioned there. I have many cameras with many zones configured. Do I have to set

```
review:
 alerts:
   required_zones:
```

for every one camera now? This is what global option seams to be made for! Please think about such behaviour.

If I can't put here zone name that is not in all my cameras, please document it somewhere. This is not related to upgrading at all.

**Reply by hawkeye217** [reply]
Yes, you must set it at the camera level, like many of the other specific parameters in the config. Frigate is working as designed here.

**Comment by luzik** [comment]
Ok, I do understand this. I just don't agree that this is logical in design then. Please rethink it again maybe.

**Reply by hawkeye217** [reply]
Consider two cameras aimed at the street. One has a `street` zone defined that captures license plates. Another has a `street` zone defined as well but captures persons. Say the user is interested in seeing people from camera 2 as alerts, but seeing license plates from camera 1 as detections. 

Setting `review:` at the global level would be ambiguous and impossible for this situation.

**Reply by luzik** [reply]
Yes! But in such cases, individual per-camera settings come into play. I have a hard time imagining a scenario where there would be an issue with assuming that when I set required_zones globally, alerts will only be triggered for the required zones. If a camera doesn't have a required zone, it won't trigger an alert. If I want a specific camera to behave differently, I can always override it in the local settings.

**Reply by hawkeye217** [reply]
That could certainly be the case, if you had the same named zone created on all of your cameras. This is where a global setting may make sense, and frigate would not give you an error. 

But to avoid ambiguity in situations like I described, this is why Frigate gives you the error message as it did.

**Reply by NickM-27** [reply]
The error message also exists because users were having typos when defining zones and then finding that the zones weren't working and didn't realize they typed it wrong.

**Reply by luzik** [reply]
Your situation can always be resolved by overriding settings at the camera level. So the main question is: **what is the most common use case for the global option** and what does the configuration logic suggest?

- Should the zone be set globally on all cameras for the option to work at all?
- Or will the option only apply to cameras that have the zone configured?

If you agree with the first option, that’s fine—I trust that Frigate is one of the best open-source projects I've seen, so you must be right. However, I have one request: Could you include information in the full configuration example indicating that if a global required_zones is set, zone must be included in all cameras? IMHO the current behavior is counterintuitive and should be documented

**Reply by NickM-27** [reply, marked as answer]
https://github.com/blakeblackshear/frigate/pull/13269
